src/main.rs

```diff
@@ -213,13 +213,13 @@ fn build_nmap_arguments<'a>(
 /// Parses the string(s) into IPs
 fn parse_addresses(opts: &Opts) -> Vec<IpAddr> {
     let mut ips: Vec<IpAddr> = Vec::new();
-    let resolver =
+    let backup_resolver =
         &Resolver::new(ResolverConfig::cloudflare_tls(), ResolverOpts::default()).unwrap();
 
     for ip_or_host in &opts.addresses {
-        match read_ips_from_file(ip_or_host.to_owned(), &resolver) {
+        match read_ips_from_file(ip_or_host.to_owned(), &backup_resolver) {
             Ok(x) => ips.extend(x),
-            _ => match parse_to_ip(ip_or_host.to_owned(), &resolver) {
+            _ => match parse_to_ip(ip_or_host.to_owned(), &backup_resolver) {
                 Ok(x) => ips.extend(x),
                 _ => {
```

Comment:
Yeah you are absolutely right. But I think this PR is not the place for this change. I'll refactor the dns resolving part of this PR. Afterwards I'll make a new PR where I will refactor all the match expressions and improve the overall style of the code. 

Would love to hear your thoughts on this. If you disagree I can add all that to this PR.

Reply:
It's up to you. I think it would be nice to not merge a PR with any "new" clippy warnings / clippy warnings that are on the diff, but you sure don't have to include these fixes here. The middle way would be to have one commit that introduces the DNS stuff and a second one that cleans up the diff, but it sure is your decision.